server/tests/controller/profile.spec.js

```diff
@@ -0,0 +1,117 @@
+import { update, getLoggedinProfile, getUserProfile } from '<controllers>/profile';
+
+const { expect } = chai;
+chai.use(chaiHttp);
+chai.use(sinonChai);
+
+const { sequelize, User } = models;
+let userId;
+
+before(async () => {
+  await sequelize.sync({ force: true });
+});
+
+afterEach(() => sinon.restore());
+
+const agent = chai.request.agent(app);
+
+describe('Update a Profile', async () => {
+  it('should update a user profile', (done) => {
+    signupUser(agent)
+      .then(() => {
+        agent
+          .patch('/api/v1/profile')
+          .send({ bio: 'Welcome to my world.' })
+          .then((res) => {
+            userId = res.body.data.id;
+            expect(res.status).to.be.equal(200);
+            expect(res.body.message).to.equal('Profile updated successfully');
+            expect(res.body.data.bio).to.equal('Welcome to my world.');
+          });
+        done();
+      })
+      .catch((err) => {
+        done(err);
+      });
+  });
+
+  it('should return server error for update controller', async () => {
+    const req = { body: { bio: 'Welcome to my world.' } };
+    const res = {
+      status() {},
+      json() {}
+    };
+    const stubStatus = sinon.stub(res, 'status').returnsThis();
+    const stubUser = sinon.stub(User, 'findByPk').throws();
+    await update(req, res);
+    expect(res.status).to.have.been.calledWith(500);
+    stubStatus.restore();
+    stubUser.restore();
+  });
+
+  it('should fetch a user\'s profile', (done) => {
+    signupUser(agent)
+      .then(() => {
+        agent
+          .get('/api/v1/profile')
+          .then((res) => {
+            expect(res.status).to.be.equal(200);
+            expect(res.body.message).to.equal('User profile');
+            expect(res.body.data.bio).to.equal('Welcome to my world.');
+          });
+        done();
+      })
+      .catch((err) => {
+        done(err);
+      });
+  });
+
+  it('should return server error for getProfile controller', async () => {
+    const req = { body: { bio: 'Welcome to my world.' } };
+    const res = {
+      status() {},
+      json() {}
+    };
+    const stubStatus = sinon.stub(res, 'status').returnsThis();
+    const stubUser = sinon.stub(User, 'findByPk').throws();
+    await getLoggedinProfile(req, res);
+    expect(res.status).to.have.been.calledWith(500);
+    stubStatus.restore();
+    stubUser.restore();
+  });
+
+  it('should return resource for getUserProfile controller', async () => {
+    const req = { body: { bio: 'Welcome to my world.' }, params: { id: userId } };
+    const res = {
+      status() {},
+      json() {}
+    };
+    const stubStatus = sinon.stub(res, 'status').returnsThis();
+    await getUserProfile(req, res);
+    expect(res.status).to.have.been.calledWith(200);
+    stubStatus.restore();
+  });
+
+  it('should return error 400 for invalid user id (uuid)', (done) => {
+    signupUser(agent)
+      .then(() => {
+        agent
+          .get('/api/v1/profile/123')
```

Comment:
A resource that does not exist should be not be found so 404

Reply:
This param is badly constructed and would throw error when jwt tries to decode it. I added another route to return 404 for a well formed token that doesn't exist.

Reply:
This could be a valid id in cases. On the frontend, I would expect a `not found` error to be thrown not an invalid uuid.